### PR TITLE
Update/0.17.0

### LINKED ISSUE #74: [bug]: When rebuilding, the Makefile does not store all build dependencies
### Prerequisites

- [X] I have searched for existing issues that already report this problem.

### Device

Laptop/Desktop

### Device OS

MacOS

### Device OS Version

latest

### Browser

Firefox

### Browser Version

104.0

### Current Behavior

rebuilds most of the .tar and .s9pk every time, if I only make changes to the actions code, it does not recognize the change with the `make` command.

### Expected Behavior

all dependencies should be saved locally

### Steps to Reproduce

`make`

### Anything else?

_No response_

### LINKED ISSUE #78: [feat]: Remove autopilot or bury it deep in advanced settings with many warnings
### Prerequisites

- [X] I have searched for existing issues that already suggest this feature.

### Describe the Feature Request

This would just stop people from having bad things happening due to issues with autopilot.

### Describe the Use Case

.

### Describe Preferred Solution

_No response_

### Describe Alternatives

_No response_

### Anything else?

_No response_

### LINKED ISSUE #97: [feat]: Improved Safety Measures for (Umbrel) Migrations
Action should refuse to run when a node already exists (LND has been run) OR should deal with existing data

### LINKED ISSUE #110: [bug] handle restore of backup made before initial config save
No reason to back up a service that has never been configured, but if the user does it, the service should be able to handle restoring it.

In this case, LND should create the start9 dir on restore if it does not exist

### PR DESCRIPTION
**Pull request by Dominion5254**
- Update upstream to v0.17.0
- Add warning to Autopilot config - closes #78 
- Create necessary directory on backup restore of unconfigured installation - closes #110 
- Throw error on migration actions if LND data already exists - closes #97 
- Add actions directory to Makefile - closes #74 
- Display Watchtower Server URL in Properties (if enabled)
- Remove add-watchtower action as the config includes the functionality to add watchtowers
- Minor fixes, refactoring, and doc updates

### CODE REVIEW

File: Dockerfile
Comment (k0gen) [inline Dockerfile:1]: Consider optimizing by:
* using `--no-cache or --update with rm -rf /var/cache/apk/*`
* you can just install `yq` in apline
* you have curl already `wget` is not needed

File: README.md [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
+## Build environment
+Before building the LND package, your build environment must be setup for building StartOS services. Instructions for setting up the proper build environment can be found in the [Developer Docs](https://docs.start9.com/latest/developer-docs/packaging).
 
 ## Dependencies
 
 - [docker](https://docs.docker.com/get-docker)
 - [docker-buildx](https://docs.docker.com/buildx/working-with-buildx/)
 - [yq (version 4)](https://mikefarah.gitbook.io/yq)
-- [embassy-sdk](https://github.com/Start9Labs/embassy-os/blob/master/backend/install-sdk.sh)
+- [start-sdk](https://github.com/Start9Labs/embassy-os/blob/master/backend/install-sdk.sh)
```
Comment (k0gen) [inline README.md:13]: use `sdk` instead of `master`

Comment (Dominion5254) [reply README.md]: Updated link

File: Dockerfile
Comment (k0gen) [inline Dockerfile:1]: Double check whether you require all these packages added by `apk add`. It's possible that some of these may be remnants from earlier builds and no longer necessary.

Comment (Dominion5254) [reply Dockerfile:1]: The optimizations you added look good. The packages in the Dockerfile were added with the 0.16.0 release to utilize the system generated cert.

**Review by k0gen — APPROVED** [review]

**Review by gStart9 — APPROVED** [review]

**Pull request merged**